src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs

```diff
@@ -126,6 +126,12 @@ INegatedPatternOperation or
                     ? ValueUsageInfo.ReadWrite
                     : ValueUsageInfo.Write;
             }
+            else if (operation.Parent is ISimpleAssignmentOperation simpleAssignmentOperation &&
+                simpleAssignmentOperation.Value == operation &&
+                simpleAssignmentOperation.IsRef)
+            {
+                return ValueUsageInfo.ReadableWritableReference;
```

Comment:
if you have: `readonly ref int x = ref y` does this count as a read/write ref now?  or just a read-ref?  It should be the latter.  Can  you check, fix if necessary, and add test?

Reply:
➡️ Filed #58466 to resolve this comment

Reply:
Yep, agreed. This case should be readable reference, I'll make the change